### PR TITLE
Submissions query api

### PR DESCRIPTION
**Pull request by Johnny850807**
Submissions query api

### CODE REVIEW

File: Spring-Boot/Spring-Boot-Submission/src/main/java/tw/waterball/judgegirl/springboot/submission/controllers/StudentSubmissionController.java [hunk 1]
```diff
@@ -62,30 +61,14 @@
                                           @RequestParam(SUBMIT_CODE_MULTIPART_KEY_NAME) MultipartFile[] submittedCodes) {
         return tokenService.returnIfGranted(studentId, authorization, token -> {
             boolean throttling = !token.isAdmin();
-            Bag bag = getBagFromHeaders(token, headers);
+            Bag bag = token.isAdmin() ? extractBagsFromHeaders(headers) : Bag.empty();
             SubmitCodeRequest request = convertToSubmitCodeRequest(problemId, langEnvName, studentId, submittedCodes, bag, throttling);
             SubmissionPresenter presenter = new SubmissionPresenter();
```
Comment (Wally5077) [inline Spring-Boot/Spring-Boot-Submission/src/main/java/tw/waterball/judgegirl/springboot/submission/controllers/StudentSubmissionController.java:66]: SubmitCodeRequest -> var
SubmissionPresenter -> var

File: Spring-Boot/Spring-Boot-Submission/src/main/java/tw/waterball/judgegirl/springboot/submission/controllers/StudentSubmissionController.java [hunk 2]
```diff
@@ -112,15 +95,17 @@ SubmissionView getSubmission(@RequestHeader("Authorization") String authorizatio
 
     @GetMapping
     List<SubmissionView> getSubmissions(@RequestHeader("Authorization") String authorization,
-                                        @RequestParam(value = "page", required = false) Integer page,
+                                        @RequestParam(required = false) Integer page,
```
Comment (Wally5077) [inline Spring-Boot/Spring-Boot-Submission/src/main/java/tw/waterball/judgegirl/springboot/submission/controllers/StudentSubmissionController.java:98]: Nice Change

File: Spring-Boot/Spring-Boot-Submission/src/main/java/tw/waterball/judgegirl/springboot/submission/controllers/StudentSubmissionController.java [hunk 3]
```diff
@@ -62,30 +61,14 @@
                                           @RequestParam(SUBMIT_CODE_MULTIPART_KEY_NAME) MultipartFile[] submittedCodes) {
```
Comment (Wally5077) [inline Spring-Boot/Spring-Boot-Submission/src/main/java/tw/waterball/judgegirl/springboot/submission/controllers/StudentSubmissionController.java:61]: delete APPLICATION_JSON_VALUE

File: Spring-Boot/Spring-Boot-Submission/src/main/java/tw/waterball/judgegirl/springboot/submission/controllers/SubmissionQueryController.java [hunk 4]
```diff
@@ -30,10 +35,40 @@ public String health() {
...
+        bagQueryParameters.remove("ascending");
+        return tokenService.returnIfAdmin(authorization, token -> {
+            var presenter = new SubmissionsPresenter();
+            if (ids != null) {
+                getSubmissionsUseCase.execute(ids, presenter);
+            } else {
+                SortBy sort = null;
+                if (sortBy != null) {
+                    sort = new SortBy(sortBy, ascending);
+                }
```
Comment (Wally5077) [inline Spring-Boot/Spring-Boot-Submission/src/main/java/tw/waterball/judgegirl/springboot/submission/controllers/SubmissionQueryController.java:61]: NIT: SortBy sort = requireNonNullElse(sortBy, new SortBy(sortBy, ascending));

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

**Review by timt1028 — APPROVED** [review]
LGTM

**Review by Wally5077 — APPROVED** [review]
LGTM

**Pull request merged**